### PR TITLE
First set of scope exercises

### PR DESCRIPTION
**Pull request by clln**
@mikesjewett Here are the first four exercises of the Scope level. Not sure who I should ping for review.

### CODE REVIEW

**Comment by mikesjewett** [comment]
@R-V-S interested in reviewing these? They'll give you a feel for exercise structure and conventions, and you can comment on the content itself as well.

**Comment by R-V-S** [comment]
Yup, looking at it now. Thanks!

**Comment by R-V-S** [comment]
Hi, sorry for the delay in getting feedback. I think it's great, but I do have some thoughts. All just my 2c. 

### 01-global-scope

> Scope determines which variables a function can access. The JavaScript scope model is known as lexical scope, where code may only access variables defined in the same block.

Concerns: 

1. Scope can be a difficult concept for new coders to understand, so a little more depth to the explanation could be helpful.
2. ES6 has block-level scope, so I wouldn't say that it's just about what a function can access. 
3. I think it's hard to understand lexical scope without also mentioning the difference between lexical and dynamic scope. It's also very important that the student does understand the difference, because of the pesky, sometimes-dynamically-scoped `this` keyword.

Here's how I would write it (though I realize I tend to be pretty verbose, so I'm not saying this is what the text should be):

---

Scope is the range in which a variable can be accessed or referenced. It can be thought of as the "visibility" of a variable, and just like real world visibility, it changes based on context. 

A variable is _in scope_ if the variable can be accessed (or "seen") within a given _context_ or _environment_. 

A variable is _out of scope_ if it can't be accessed, in which case attempting to access it will result in an error – JavaScript will act as if the variable doesn't exist. Javascript can't "see" the variable from within that context.

There are two types of scope: 

1. **Lexical scope**: The scope of a variable is determined by where in the code it's defined.
2. **Dynamic scope**: The scope of a variable is determined by the order of execution, e.g. which function is executing the code. 

With one notable exception (the **this** keyword - to be covered later), JavaScript variables follow lexical scope. 

---
> There are two types of scope: global and local. Variables declared in the global scope, outside of any functions, are properties of the global object to which all functions in the script have access. For example:

Concern: I wouldn't say that global and local are _types_ of scope, but rather categories of contexts. Without explaining how context is defined, I think that a deep understanding of scope may be hard to grasp.

Here's how I would write it:

---
A scope is confined to a context, and there are three types of context in JavaScript:

1. **Global**: The global context exists everywhere in a codebase, and variables defined _globally_ can be accessed from anywhere.
2. **Function**: Each function contains its own context. Variables defined within a function can't be directly accessed outside of that function. They are _local_ to the function.
3. **Block**: If a variable is defined with the `let` statement (an ES6 feature), it can't be accessed outside of the block of code it exists in. They are _local_ to the block.  
  A block of code is a portion of code contained within curly braces, and apply not just to functions, but also to loops, switch statements, and stand-alone curly braces.

---

### 02-local-scope

I think it might be worth mentioning block scope and the _let_ statement. 

---
tl;dr:

1. It might not be a bad idea to add an introduction to scope before going into global scope. 
2. I think it's important for coders to understand the concepts of _dynamic scope_ and _context_ (or _environment_). An understanding of context/environment is especially helpful in understanding closures.
3. It might be worth adding an exercise (or two) on dynamic scope and the _this_ keyword, unless that's covered elsewhere.
4. I think block scope and the _let_ statement should be covered. Although it has no browser support (yet), it can be used in node and babel today.

Anyway, great job & I'm looking forward to seeing this go live.

_Cheers,_  
Raj
